server/jobs/job_manager.go

```diff
@@ -58,47 +57,20 @@ func NewJobManager(papi cluster.JobPluginAPI, env mscalendar.Env) *JobManager {
 // AddJob accepts a RegisteredJob, stores it, and activates it if enabled.
 func (jm *JobManager) AddJob(job RegisteredJob) {
 	jm.registeredJobs.Store(job.id, job)
-}
-
-// OnConfigurationChange activates/deactivates jobs based on their current state, and the current plugin config.
-func (jm *JobManager) OnConfigurationChange(env mscalendar.Env) error {
-	jm.mux.Lock()
-	defer jm.mux.Unlock()
-	jm.env = env
-
-	jm.registeredJobs.Range(func(k interface{}, v interface{}) bool {
-		job := v.(RegisteredJob)
-		enabled := (job.isEnabledByConfig == nil) || job.isEnabledByConfig(env)
-		_, active := jm.activeJobs.Load(job.id)
-
-		// Config is set to enable. Job is inactive, so activate the job.
-		if enabled && !active {
-			err := jm.activateJob(job)
-			if err != nil {
-				jm.env.Logger.Warnf("Error activating %s job. err=%v", job.id, err)
-			}
-		}
-
-		// Config is set to disable. Job is active, so deactivate the job.
-		if !enabled && active {
-			err := jm.deactivateJob(job)
-			if err != nil {
-				jm.env.Logger.Warnf("Error deactivating %s job. err=%v", job.id, err)
-			}
-		}
-
-		return true
-	})
-	return nil
+	err := jm.activateJob(job)
+	if err != nil {
+		jm.env.Logger.Warnf("Error activating %s job. %v", job.id, err)
+	}
 }
 
 // Close deactivates all active jobs. It is called in the plugin hook OnDeactivate.
 func (jm *JobManager) Close() error {
+	jm.env.Logger.Debugf("Deactivating all jobs due to plugin deactivation.")
 	jm.activeJobs.Range(func(k interface{}, v interface{}) bool {
 		job := v.(*activeJob)
 		err := jm.deactivateJob(job.RegisteredJob)
 		if err != nil {
-			jm.env.Logger.Debugf("Failed to deactivate job: %v", err)
+			jm.env.Logger.Warnf("Failed to deactivate %s job: %v", job.id, err)
 		}
```

Comment:
@lieut-data We are experiencing an issue testing the cluster jobs in HA. We are testing with a fresh cloud server with `--size miniHA`, which spins up a cluster with two nodes.

I will be referring to the jobs as a singular "job" (`cluster.Job`), as all the jobs are showing the same behavior. The job activates properly on plugin activation. The successful activation message is logged by both nodes by the `jm.activateJob` method. The job runs its first execution with no issues. However, the job does not run again when its configured interval comes around.

Disabling the plugin results in the call to `job.Close()` to hang. Tracing the logs provided by this PR, the `Deactivating all jobs due to plugin deactivation` message is logged once (due to the node blocking on disable, and not allowing the second node to begin disabling), and does not log the debug message in `jm.deactivateJob`, nor the warning in this function. Therefore the `job.Close()` method must be blocking.

I'm not sure where to begin with debugging this. I'd like to debug locally, but I haven't set up an HA environment on my local machine. How do you advise moving forward with this?

Reply:
To be clear, this behavior was occurring before this PR. Part of the reason for this PR is to simplify the logic, to minimize the surface area for errors like the ones described above to occur.

Reply:
(For anyone following this thread, see https://community-daily.mattermost.com/core/pl/auyr1cd1q7nu9ytmtb9a5fod8a for the resolution.)